### ISSUE TITLE
Post sucessful start up script

### ISSUE THREAD

**Issue by TGWolf** [fail2ban/fail2ban#2874]
I am currently using fail2ban 0.10.2-2.1 on Debian 10.

I have added in  custom option to send notifications to slack which works fine, but I would like a way to execute a command 'after' the jail has restart and all the bans are in place. I have norestored set to 1 as I do not want to get spammed with all the reban fortifications on a restart.

Within my custom ruby gem I have an option for 'restarted' which sends a slack notification telling me all the jails I have active and the count of the rules in place, and I would like to be able to call this after a start/restart once the bans are in place so I can see how many bans are in place.

All of my code is in place but I cannot find anyway to hook into the start/restart process in order to call the script.

**Comment by sebres** [comment]
There is no such (post-start) action command which will be called if all the bans are there.
At the moment you can only use:
- `actionstart` to signal jail is started, executed before first ban occurs if `actionstart_on_demand = true` (default) or if `actionstart_on_demand = false` if jail gets started.
- `actionreload` to signal jail gets reloaded (executed by reload only, but not by restart)

**Comment by TGWolf** [comment]
I have come up with a work around for now. I have added a systemd drop-in file for fail2ban, 

Drop in:
[Service]
ExecStartPost=/etc/fail2ban/action.d/slack-restart-notifier

Script:
#!/usr/bin/env bash

(sleep 5; /etc/fail2ban/action.d/slack-notify -a restarted)&

exit 0


Its a bit of a hack, but it does what I need, as all the jails are loaded

**Comment by sebres** [comment]
Revisited that. It is not so trivial as one can think - the action if IP gets banned will be executed asynchronously in different thread, so if N tickets are restored after restart, they are not yet banned immediately and as long as they will be processed fail2ban can find new failures and ban them also (so the question whether restart is fulfilled is not simple).

Just curios - why it is important to know that all bans are restored after jail start?
I mean where do you see a difference between this 2 variants basically:

Variant 1 | Variant 2
-|-
3 active bans found | 3 active bans found
3 bans restored | 2 bans restored
3 tickets banned | 30 failures found in log
restart is complete? | 2 tickets banned
30 failures found in log | 10 tickets queued to ban
10 tickets queued to ban | 11 tickets banned
10 tickets banned | restart is complete?

In both cases fail2ban restoring 3 active tickets (and found 10 others to ban), so as result 13 IPs going banned, just the timing is different.
Where is the proper timepoint for such an post start-up command?
For which purposes exactly?

**Comment by TGWolf** [comment]
The reason I do it is because the information is sent to a security related slack channel for my company, when we ban/unban and IP we send the following info:

Action
Hostname
IP Address
Country
AS Owner
Jail
Current Blocks

When we restart we just want to know how many ips are currently blocked hence the hacky work around we put in place. It probably isn't useful/relevant to most people which is why I was happy to just use the method I detailed.

This is partly due to the management of the machines being in puppet and sometimes a service restart is required for config changes, and thats where we were missing info. fail2ban-client restart works fine but a service fsil2ban restart is where we found an issue.

I am happy that the solution we have adopted meets our needs and it is such a bespoke case that it isnt probably worth the effort trying to make it a built in solution.
